dot/core/messages_test.go

```diff
@@ -2,3 +2,265 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 package core
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/peerset"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	mocksruntime "github.com/ChainSafe/gossamer/lib/runtime/mocks"
+	"github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/golang/mock/gomock"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	errBestHeader = errors.New("best header error")
+	errGetRuntime = errors.New("get runtime error")
+	errTrieState  = errors.New("trie state error")
+)
+
+// TODO use new test format
+
+func TestService_TransactionsCount(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockTxnStateEmpty := NewMockTransactionState(ctrl)
+	mockTxnState := NewMockTransactionState(ctrl)
+
+	txs := make([]*transaction.ValidTransaction, 2)
```

Comment:
meganit just for a bit more expliciteness since we never set it.

```suggestion
	txs := []*transaction.ValidTransaction{nil, nil}
```